wardemo/src/main/java/org/jfclere/tomcatPI/FrameBuffer.java

```diff
@@ -23,114 +23,101 @@
      * @throws FileNotFoundException 
      */
     public FrameBuffer() throws FileNotFoundException {
-        // TODO Auto-generated constructor stub
     	pi = new PIFrameBuffer("/dev/fb1", "rw");
         try {
             pi.clear(0);
         } catch(java.io.IOException ex) {
         }
     }
-
+    
 	/**
 	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
 	 */
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-
         // Process paramters.
-        String si = request.getParameter("i");
-        String sj = request.getParameter("j");
-        if (sj != null && si != null) {
-            int i = Integer.parseInt(si);
-            int j = Integer.parseInt(sj);
-            pi.writepix(i, j, pi.color(0,63,0));
+        String param_i = request.getParameter("i");
+        String param_j = request.getParameter("j");
+        String param_red = request.getParameter("red");
+        String param_green = request.getParameter("green");
+        String param_blue = request.getParameter("blue");
+
+        if (param_j != null && param_i != null) {
+            int p_i = Integer.parseInt(param_i);
+            int p_j = Integer.parseInt(param_j);
+            int red = Integer.parseInt(param_red);
```

Comment:
What happens is red, green or blue are null?

Reply:
Well,  this probably is a better solution but they shouldn't be null at any time as the JavaScript which send the request is triggered only when a color (so a RGB value) is selected.

to be moresecure/clean i would change the following things in the **script.js** : 

  ```javascript

  function getRed(rgb_str) {
    var red = 0;
    var rgb_red = /rgba?\(\s*(\d+),(?:[\d\s]*),(?:[\d\s]*)(?:,\s*[\d\.]+)?\)/.exec(rgb_str);
    if(rgb_red) {
       red = rgb_red;
    }

    return red;
  }

  function getGreen(rgb_str) {
    var green = 0;
    var rgb_green = /rgba?\((?:[\d\s]*),\s*(\d+),(?:[\d\s]*)(?:,\s*[\d\.]+)?\)/.exec(rgb_str);
    if(rgb_green) {
       green = rgb_green;
    }

    return green;
  }

  function getBlue(rgb_str) {
    var blue = 0;
    var rgb_blue = /rgba?\((?:[\d\s]*),(?:[\d\s]*),\s*(\d+)(?:,\s*[\d\.]+)?\)/.exec(rgb_str);
    if(rgb_blue) {
       blue = rgb_blue;
    }

    return blue;
  }


  ```

  and in FrameBuffer.java :
  
  ```java
    if (param_j != null && param_i != null && param_red != null && param_green != null && param_blue != null) {
        int p_i = Integer.parseInt(param_i);
        int p_j = Integer.parseInt(param_j);
        int red = Integer.parseInt(param_red);
        int green = Integer.parseInt(param_green);
        int blue = Integer.parseInt(param_blue);

        if(p_i >= 0 && p_i <= 7 && p_j >= 0 && p_j <= 7 && pi.color(red, green, blue)) {
            pi.writepix(p_i, p_j, pi.color(red, green, blue));
        }
    }
  
  ```

Reply:
+1